base_layer/contacts/src/chat_client/src/client.rs

```diff
@@ -81,17 +82,28 @@ impl Drop for Client {
 }
 
 impl Client {
-    pub fn new(identity: Arc<NodeIdentity>, config: ApplicationConfig, user_agent: String) -> Self {
+    pub fn new(
+        identity: Arc<NodeIdentity>,
+        address: TariAddress,
```

Comment:
In my head, the additions to the address would have somehow integrated back into the NodeIdentity, as opposed to needing to exist outside of it. Is there something stopping us from these constructs being put together? 
NodeIdentity has the comms public key. TariAddress *was* a combination of the comms public key + network. We've made the addition of the view and spend key. But could we keep those within the NodeIdentity as well? 

I realize there's probably reasons not to, but they aren't immediately apparent to me.

Reply:
The biggest reason not to, is where they live. Node identity lives inside the comms layer and is aimed at network identity. Base nodes don't have a vire/spend key, they only have a comms key.